sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -801,7 +908,10 @@ class CodegenContext {
         addNewFunction(name, code)
       }
 
-      foldFunctions(functions.map(name => s"$name(${arguments.map(_._2).mkString(", ")})"))
+      val exprs = transformFunctions(functions.map(name =>
+        s"$name(${arguments.map(_._2).mkString(", ")})"))
+
+      splitExpressions(exprs, funcName, arguments)
```

Comment:
Changes I made here to `splitExpressions` were to handle instances where the split code method references were still over 64kb. It would seem this problem is addressed by @mgaido91 in #19480, and that implementation is much more thorough, so if that PR gets merged, I'd prefer to rebase against that.